src/support/topological_orders.h

```diff
@@ -78,19 +88,86 @@ struct TopologicalOrders {
 
     // Select the next available vertex, decrement in-degrees, and update the
     // sequence of available vertices. Return the Selector for the next vertex.
-    Selector select(TopologicalOrders& ctx);
+    Selector select(TopologicalOrders& ctx, bool minOrder);
 
     // Undo the current selection, move the next selection into the first
     // position and return the new selector for the next position. Returns
     // nullopt if advancing wraps back around to the original configuration.
     std::optional<Selector> advance(TopologicalOrders& ctx);
   };
 
+  void pushChoice(size_t);
+  size_t popChoice();
+
   // A stack of selectors, one for each vertex in a complete topological order.
   // Empty if we've already seen every possible ordering.
   std::vector<Selector> selectors;
 };
 
+// A utility for finding a single topological order of a graph.
+struct TopologicalSort : private TopologicalOrders {
+  TopologicalSort(const std::vector<std::vector<size_t>>& graph)
+    : TopologicalOrders(graph, false) {}
+
+  const std::vector<size_t>& operator*() const {
+    return TopologicalOrders::operator*();
+  }
+};
```

Comment:
This constructor is identical to the parent, and this operator doesn't seem to modify anything either - I am sure I'm missing something?

Reply:
Nope, this is providing a small subset of the functionality of the superclass, packaged under a different name that is more appropriate for the common case of needing just one topological sort.

Reply:
Ah, thanks... I was missing the word "private" on line 108. Makes sense now.